tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/xml/XmlParserHelper.java

```diff
@@ -120,7 +121,11 @@ public <T> T getJaxbRoot(InputStream stream, Class<T> clazz, Schema schema)
 		ContextProvidingValidationEventHandler handler = new ContextProvidingValidationEventHandler();
 		try {
 			staxEventReader = new JpaNamespaceTransformingEventReader( staxEventReader );
-			JAXBContext jaxbContext = JAXBContext.newInstance( ObjectFactory.class );
+
+			ClassLoader cl = NullnessUtil.castNonNull( ObjectFactory.class.getClassLoader() );
+			String packageName = NullnessUtil.castNonNull( ObjectFactory.class.getPackage() ).getName();
+			JAXBContext jaxbContext = JAXBContext.newInstance( packageName, cl );
```

Comment:
```suggestion
			// Class.getClassLoader() may return null if the class was loaded by the bootstrap class loader,
			// but since we don't expect the annotation processor to be loaded by that class loader, 
			// we expect the return to be non-null and hence cast
			ClassLoader cl = NullnessUtil.castNonNull( ObjectFactory.class.getClassLoader() );
			String packageName = NullnessUtil.castNonNull( ObjectFactory.class.getPackage() ).getName();
			JAXBContext jaxbContext = JAXBContext.newInstance( packageName, cl );
```

Reply:
Comment added 👍